### PR TITLE
Janus webRTC gateway module

### PR DESCRIPTION
**Pull request by ppp0**
Installation
- Janus gateway (https://github.com/meetecho/janus-gateway) / modular-transport branch
- Daemon https://janus.conf.meetecho.com/docs/service.html
- Monit
- Bipbip

Compiling
- we need "--enable-post-processing"

Configuration:
- janus core
- plugins class
- transport class

### CODE REVIEW

File: modules/janus/manifests/deps/libsrtp.pp [hunk 1]
```diff
@@ -0,0 +1,11 @@
+class janus::deps::libsrtp(
+  $version = '1.5.0',
+  $build_tests = false,
+) {
+
+  helper::script { 'install libsrtp':
+    content => template("${module_name}/deps/libsrtp_install.sh"),
+    unless  => " ${version}$'",
```
Comment (ppp0) [inline modules/janus/manifests/deps/libsrtp.pp]: not clear how to check the version.. :/ - May be just omit the param?

Comment (kris-lab) [inline modules/janus/manifests/deps/libsrtp.pp]: What about `pkg-config --modversion libsrtp`

Comment (ppp0) [inline modules/janus/manifests/deps/libsrtp.pp]: Works if one has `pkg-config` installed BEFORE building the library. Thank you

**Comment by ppp0** [comment]
Hm I am not completely happy with the specs.. How to properly validate `wss://` ?
@kris-lab @njam @tomaszdurka ideas?

**Comment by kris-lab** [comment]
Just a few thoughts:
- janus: I would not support SSL for now?
- janus: make admin-port, websocket-port + bind interface configurable?
- janus: make token, secret configurable
- janus: make turn/stun server configurable?
- janus: make plugins enable/disable configurable?
- janus::service: add bipbip ?
- janus:plugin: maybe we should add some plugins abstraction for our custom plugins and janus one?

@ppp0 let's pair review on this?

**Comment by kris-lab** [comment]
Also maybe all dependencies should be as separate modules? I mean `srtp`, `usrsctp`, `websockets`?

**Comment by ppp0** [comment]
About the modules: right now, the dependencies are separate classes (`libsrtp`, `libusrsctp`, `libwebsockets`).. If we go make modules out of them we should be aware that they build and provide libraries for further compilation, do may be make them a subclass of  `build` similar to `build::dev::zlib1g` ? Not sure what is more consistent. Otherwise leave it as dep fro Janus as long as they're not depended upon anywhere else

**Comment by kris-lab** [comment]
Agree on dependencies. I was following `x264`, `yasm` etc. but there are binaries apps and not libs only. Let's keep it in Janus then!

**Comment by kris-lab** [comment]
As discussed with @ppp0 
- we should stick to janus `modular-transport` branch
- we should remove SSL as not needed (use default provided by Janus)
- abstract `transport` into classes?
- abstract `plugin` into classes? Disable current plugins! Enable custom plugins

plus copying myself
> janus: I would not support SSL for now?
janus: make admin-port, websocket-port + bind interface configurable?
janus: make token, secret configurable
janus: make turn/stun server configurable?
janus: make plugins enable/disable configurable?
janus::service: add bipbip ?
janus:plugin: maybe we should add some plugins abstraction for our custom plugins and janus one?

**Comment by ppp0** [comment]
>- we should stick to janus `modular-transport` branch

OK

>- we should remove SSL as not needed (use default provided by Janus)

OK

>- abstract `transport` into classes?

OK

>- abstract `plugin` into classes? Disable current plugins! Enable custom plugins

OK - current plugins almost completely deconfigured from the build

> janus: I would not support SSL for now?

OK - support is built-in - just need to provide certificate+key

>janus: make admin-port, websocket-port + bind interface configurable?

OK (see transport configurables)

>janus: make token, secret configurable

to be done

>janus: make turn/stun server configurable?

to be done

> janus: make plugins enable/disable configurable?

as-of-now one can disable plugins in the main config fro janus

>janus::service: add bipbip ?

good idea

>janus:plugin: maybe we should add some plugins abstraction for our custom plugins and janus one?

only our custom plugins at the moment but ready to be expanded if needed

**Comment by ppp0** [comment]
>janus: make token, secret configurable

OK
>janus: make turn/stun server configurable?

OK
>janus::service: add bipbip ?

May be with a plugin? https://github.com/cargomedia/bipbip/issues/132

Otherwise, @kris-lab please review

**Comment by ppp0** [comment]
Hm, we need to write better spec tests for plugin building and installation

**Comment by kris-lab** [comment]
- audioroom: config file should be update; it is not empty anymore [see](https://github.com/cargomedia/janus-gateway-audioroom/blob/master/conf/janus.plugin.cm.audioroom.cfg.sample)
- plugin:audioroom : params should be adjusted according to the config file
- templates:deps:plugin:libinstall.sh : is it good place and name for plugin installer? wasn't it better to move it into `templates/install-plugin.sh`

rest look good

**Comment by ppp0** [comment]
* Config for audioroom is updated
* Agree with refactoring plugin building (see also janus_version enforcement)
* Centralised janus version number

@kris-lab please re-review

**Comment by kris-lab** [comment]
lgtm!

**Comment by ppp0** [comment]
thanks for pair-reviewing!

**Pull request merged**